### ISSUE TITLE
ArrayTypeHandler can't get value unless I refer it in resultMap 

### ISSUE THREAD

**Issue by q975144677** [mybatis/spring-boot-starter#496]
When i select an array in postgre ,ArrayTypeHandler can't get the value ,
Unless I refer the ArrayTypeHandler in resultMap。
I try to debug，,i find the fact that mybatis will invoke ArrayTypeHandler twice ,and in the first invoke， resultSet.getArray is null ,but the second time it get the array value 。
If I dont refer the handler explicitly， mybatis will call the handler only once，so the bean cant get the value。
I want know the reason why it happen and how can i get the array value without refer or create the typeHandler , thanks

**Comment by kazuki43zoo** [comment]
@q975144677 

Probably this problem not belong this project. Please create new issue to the [mybatis-core module](https://github.com/mybatis/mybatis-3). Please attach a repro project if possible.
